src/Controller/UserController.php

```diff
@@ -80,17 +159,30 @@ public function add(): ?string
             }
 
             $user = array_map('trim', $_POST);
-            // TODO validations (length, format...)
-
+            $user = $this->cleanParam($user);
+            if ($user) {
+                $errors = $this->checkData($user);
+            } else {
+                $errors[] = "Entreè(s) avec format invalide";
+            }
             // if validation is ok, insert and redirection
             $userManager = new UserManager();
-            $id = $userManager->insert($user);
+            if (empty($errors)) {
+                try {
+                    $id = $userManager->insert($user);
+                } catch (PDOException $err) {
```

Comment:
try to put all PDO Exceptions (and try catch) inside manager class since it is related to database connexion